src/engraving/libmscore/check.cpp

```diff
@@ -232,15 +234,16 @@ void Measure::fillGap(const Fraction& pos, const Fraction& len, track_idx_t trac
          len.numerator(), len.denominator(),
          stretch.numerator(), stretch.denominator(),
          track);
-    TDuration d;
-    d.setVal(len.ticks());
-    if (d.isValid()) {
+    std::vector<TDuration> durationList = toRhythmicDurationList(len, true, pos, score()->sigmap()->timesig(tick()).nominal(), this, 0);
```

Comment:
Shouldn't `score()->sigmap()->timesig(tick()).nominal()` be replaced with `_len`, to take irregular measures into account?

Reply:
I am thinking the answer is no, at least not so directly.  For example, a measure of 4/4 might be lengthened to 9/8 to accommodate some sort of cadenza, but that doesn't mean we should use the group rules for compound meter.

Or, another case - you might have a measure of length 4/4 that has been split into two measures of 5/8 and 3/8, but that doesn't mean you want leading rests to actually group according to those time signatures.  They should still group as 4/4.  But, the first partial measure should group as if it is the *first* part of a 4/4 measure, and the second partial measure should group as if it is the *second*.  So for example, if there is to be one beat worth of rest, the first 5/8 portion should show it as a quarter since this truly is beat 1, but the second 3/8 portion should show it as two eighths since this is actually starting on the "&" of 3.

Unfortunately, we don't have a way of knowing whether any given irregular measure is meant to be the first portion or second portion of a split measure, or if indeed it represents a split at all or something else entirely.

The swing playback system has to deal with similar issues of trying to figure out if a shorter-than-nominal measure that has odd number of eighths should be seen as starting on the beat or off the beat.  We used an awkward but somewhat effective hack, basically checking to see if this is the first measure of the piece (making it a pickup) and also if there is a system or similar break delineating the partial measures.  Probably we should be checking for repeats too.

Might be nice to have a general purpose function to determine this, and to do it a bit smarter, like looking behind and ahead for a "matching" partial measure.  Such a function would also come in handy when generating rests as well as when determining beaming for pickup measures, which we basically always get wrong since we treat them as "first portion" rather than "second portion".

Reply:
Since the grouping may not be not evident, one solution is for a measure to have a grouping flag to provide, to the method filling the measure, a hint of "left" or "right". This flag might also be useful when generating flags on notes in that measure (e.g. five eighth notes in a 5/8 partial measure).

This problem also applies to a pickup measure before the first full measure.  MU typically groups a pickup measure from the left instead of from the right.  For example, a 3/8 pickup measure, before a 4/4 full measure, should be initialized as an eighth rest followed by a quarter rest, but MU defaults to quarter then eighth.